x/poa/keeper/keeper.go

```diff
@@ -199,25 +199,12 @@ func (k Keeper) ExecuteRemoveValidator(ctx sdk.Context, validatorAddress string)
 	}
 	denom := params.BondDenom
 
+	accAddress := sdk.AccAddress(valAddress)
+
 	// Check if address has some balance in bank and withdraw in case of having
 	balance := k.bk.GetBalance(ctx, accAddress, denom)
-	if balance.IsPositive() {
-		coins := sdk.NewCoins(balance)
-		err = k.bk.SendCoinsFromAccountToModule(ctx, accAddress, types.ModuleName, coins)
-		if err != nil {
-			// Fail hard if we can't send coins to the module account
-			return err
-		}
-
-		err = k.bk.BurnCoins(ctx, types.ModuleName, coins)
-		if err != nil {
-			// Fail hard if we can't burn coins
-			return err
-		}
-	}
 
 	// If address also has a validator, we need to check additional conditions
-	valAddress := sdk.ValAddress(accAddress)
 	validator, err := k.sk.GetValidator(ctx, valAddress)
 	if err != nil {
 		ctx.Logger().Warn("Error getting validator", "error", err)
```

Comment:
The following check is unnecessary, if validator is not found return a custom error